src/domain/usecases/ListDanglingDocumentsUseCase.ts

```diff
@@ -1,11 +1,25 @@
 import { UseCase } from "../../webapp/CompositionRoot";
 import { NamedRef } from "../entities/Ref";
-import { InstanceRepository } from "../repositories/InstanceRepository";
+import { DocumentRepository } from "../repositories/DocumentRepository";
+import _ from "lodash";
+import { getUrls } from "../../utils/urls";
+import { extractUids } from "../../data/utils/uid";
 
 export class ListDanglingDocumentsUseCase implements UseCase {
-    constructor(private instanceRepository: InstanceRepository) {}
+    constructor(private documentRepository: DocumentRepository) {}
 
-    public async execute(): Promise<NamedRef[]> {
-        return this.instanceRepository.listDanglingDocuments();
+    public async execute(data: unknown): Promise<NamedRef[]> {
```

Comment:
We should avoid using `unknown`. We could use the function `getUrls` before the use case and and therefore we could type the `data: string[]` afterwards

Reply:
A use case should get entities, so I'd keep getUrls in the useCase and be explicit of the types that data can be (an array of union of 3 types, it seems)